code/modules/reagents/chemistry/recipes/fermi.dm

```diff
@@ -0,0 +1,174 @@
+/datum/chemical_reaction/
+	mix_sound = 'sound/effects/bubbles.ogg'
+
+	//use fermicalc.py to calculate how the curves look(i think)
+//Called for every reaction step
+/datum/chemical_reaction/fermi/proc/FermiCreate(holder)
+	return
+
+//Called when reaction STOP_PROCESSING
+/datum/chemical_reaction/proc/FermiFinish(datum/reagents/holder, var/atom/my_atom, reactVol)
+	if(clear_conversion == REACTION_CLEAR_IMPURE | REACTION_CLEAR_INVERSE)
+		for(var/id in results)
+			var/datum/reagent/R = my_atom.reagents.has_reagent("[id]")
+			if(R.purity == 1)
+				continue
+
+			var/cached_volume = R.volume
+			if(clear_conversion == REACTION_CLEAR_INVERSE && R.InverseChem)
+				if(R.InverseChemVal > R.purity)
+					my_atom.reagents.remove_reagent(R.type, cached_volume, FALSE)
+					my_atom.reagents.add_reagent(R.InverseChem, cached_volume, FALSE, other_purity = 1)
+
+			else if (clear_conversion == REACTION_CLEAR_IMPURE && R.ImpureChem)
+				var/impureVol = cached_volume * (1 - R.purity)
+				my_atom.reagents.remove_reagent(R.type, (impureVol), FALSE)
+				my_atom.reagents.add_reagent(R.ImpureChem, impureVol, FALSE, other_purity = 1)
+				R.cached_purity = R.purity
+				R.purity = 1
+
+//Called when temperature is above a certain threshold, or if purity is too low.
+/datum/chemical_reaction/fermi/proc/FermiExplode(datum/reagents, var/atom/my_atom, volume, temp, pH, Exploding = FALSE)
+	if (Exploding)
+		return
+
+	if(!pH)//Dunno how things got here without a pH, but just in case
+		pH = 7
+	var/ImpureTot = 0
+	var/turf/T = get_turf(my_atom)
+
+	if(temp>500)//if hot, start a fire
+		switch(temp)
+			if (500 to 750)
+				for(var/turf/turf in range(1,T))
+					new /obj/effect/hotspot(turf)
+
+			if (751 to 1100)
+				for(var/turf/turf in range(2,T))
+					new /obj/effect/hotspot(turf)
+
+			if (1101 to 1500) //If you're crafty
+				for(var/turf/turf in range(3,T))
+					new /obj/effect/hotspot(turf)
+
+			if (1501 to 2500) //requested
+				for(var/turf/turf in range(4,T))
+					new /obj/effect/hotspot(turf)
+
+			if (2501 to 5000)
+				for(var/turf/turf in range(5,T))
+					new /obj/effect/hotspot(turf)
+
+			if (5001 to INFINITY)
+				for(var/turf/turf in range(6,T))
+					new /obj/effect/hotspot(turf)
+
+
+	message_admins("Fermi explosion at [T], with a temperature of [temp], pH of [pH], Impurity tot of [ImpureTot].")
+	log_game("Fermi explosion at [T], with a temperature of [temp], pH of [pH], Impurity tot of [ImpureTot].")
```

Comment:
Please add the JUMP buttons and the coordinates to these logs.
This isn't really useful for anything since it literally just gives the name of the turf.